docs/external-node.md

```diff
@@ -0,0 +1,569 @@
+port, and finally output from the uplink, and the inbound traffic enters OVS
+from the uplink and output to the internal port. The IP packet is processed by
+the OpenFlow pipeline, and the non-IP packet is forwarded directly.
+
+The following diagram depicts the OVS bridge and traffic forwarding on an
+external Node:
+![Traffic On ExternalNode](assets/traffic_external_node.svg)
+
+## RBAC for antrea-agent
+
+An external Node is regarded as an untrusted entity on the network. To follow
+the principle of the least privilege is given, the RBAC configuration for
+`antrea-agent` running on `ExternalNode` feature is as follows:
+
+- Only `get`, `list` and `watch` permissions are given on resource `ExternalNode`
+- Only `update` permission is given on resource `antreaagentinfos`, and `create`
+  permission is moved to `antrea-controller`
+- for more details please refer to [vm-agent-rbac.yml](/build/yamls/externalnode/vm-agent-rbac.yml)
+
+`antrea-agent` reports its status by updating the `antreaagentinfo` resource
+which is created with the same name as the `ExternalNode`. `antrea-controller`
+creates an `antreaagentinfo` resource for each new `ExternalNode`, and then
+`antrea-agent` updates it every minute with its latest status. `antreaagentinfo`
+is deleted by `antrea-controller` when the `ExternalNode` is deleted.
+
+## Apply Antrea NetworkPolicy to ExternalNode
+
+### Antrea NetworkPolicy configuration
+
+An Antrea NetworkPolicy is applied to an `ExternalNode` by providing an
+`externalEntitySelector` in the `appliedTo` field. **The `ExternalEntity`
+resource is automatically created for each interface of an `ExternalNode`**,
+which are used by `antrea-controller` to calculate the NetworkPolicies, and
+`antrea-agent` running on the `ExternalNode` receives the internal Antrea
+NetworkPolicy objects.
+
+Following types of (from/to) network peers are supported in an Antrea
+NetworkPolicy applied to an external Node:
+
+- ExternalEntities selected by an `externalEntitySelector`
+- An `ipBlock`
+- A FQDN address in an egress rule
+
+Following actions are supported in an Antrea NetworkPolicy applied to an
+external Node:
+
+- Allow
+- Drop
+- Reject
+
+Below is an example of applying an Antrea NetworkPolicy to the external Nodes
+labeled with `role=db` to reject SSH connections from IP "172.16.100.5" or from
+other external Nodes labeled with `role=front`:
+
+```yaml
+kind: NetworkPolicy
+metadata:
+  name: anp1
+  namespace: vm-ns
+spec:
+  priority: 9000.0
+  appliedTo:
+    - externalEntitySelector:
+        matchLabels:
+          role: db
+  ingress:
+    - action: Reject
+      ports:
+        - protocol: TCP
+          port: 22
+      from:
+        - externalEntitySelector:
+            matchLabels:
+              role: front
+        - ipBlock:
+            cidr: 172.16.100.5/32
+```
+
+### Bypass Antrea NetworkPolicy
+
+In some cases, the special IP traffic is expected to bypass the Antrea
```

Comment:
"users would desire particular traffic to bypass Antrea NetworkPolicy rules on an external Node"

Reply:
done.